### ISSUE TITLE
"each" helper function leads to performance problems

### ISSUE THREAD

**Issue by marcushg** [rendrjs/rendr-handlebars#15]
TL;DR: Each time the rendr-helpers are registered as handlebars helpers, the "each"-function will save an incorrect reference to the original handlbars-"each" helper and will call another instance of itself instead -- which will result in performance degradation and more memory consumption over time. It's especially bad since we register the handlebars-helpers whenever we create a new rendr-app which in turn get's created on every request....

Long story:
The shared/helpers.js module function returns a new object containing the pre-bundled rendr-handlebars-helpers. One of those is the "each"-helper which overrides the original handlebars-helper to add rendr-specific objects to the handlebars context. lt preserves the original functionality through calling the original handlebars "each" function at the end, by saving a reference to the original function in line 7 of shared/helpers.js.
It get's problematic when we register the rendr-bundled helpers multiple times throughout our node process which happens whenever we create new rendr-app instances -- which currently happens on every request.
The first call to registerHelpers() correctly saves the Handlebars.helpers.each function in the "oldEach" variable on line 7. Then by means of the Handlebars.registerHelper() function "Handlebars.helpers.each" will now reference our "each" implementation. Calls to our "each"-helper will now correctly call the original Handlebars.helpers.each function:

rendr-helpers0.each() calls Handlebars.helpers.each()

Every subsequent call to register our helpers will now incorrectly save our overridden each-function as the "oldEach" - since Handlebars.helpers.each has been overWRITTEN by Handlebars.registerHelper() with our implementation. The result's are longer and longer callstacks:

rendr-helpers1.each() calls (rendr-helpers0.each() calls Handlebars.helpers.each())

Et cetera.... this get's linearly worse on every call to our registerHelpers(), i.e. on every request :-/

Crude fix: Save the original Handlebars.helpers.each function only once and within the module itself, not within the module.exports function. See pull request.
